pkg/artifactory/resource/webhook/resource_artifactory_webhook_base.go

```diff
@@ -82,3 +94,68 @@ var baseWebhookBaseSchema = func(webhookType string) map[string]*schema.Schema {
 		},
 	}
 }
+
+
+var baseWebhookBaseSchemaV2 = func(webhookType string) map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"key": {
+			Type:             schema.TypeString,
+			Required:         true,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.All(validation.StringLenBetween(2, 200), validation.StringDoesNotContainAny(" "))),
+			Description:      "Key of webhook. Must be between 2 and 200 characters. Cannot contain spaces.",
+		},
+		"description": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringLenBetween(0, 1000)),
+			Description:      "Description of webhook. Max length 1000 characters.",
+		},
+		"enabled": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: "Status of webhook. Default to 'true'",
+		},
+		"event_types": {
+			Type:     schema.TypeSet,
+			Required: true,
+			MinItems: 1,
+			Elem:     &schema.Schema{Type: schema.TypeString},
+			Description: fmt.Sprintf("List of Events in Artifactory, Distribution, Release Bundle that function as the event trigger for the Webhook.\n"+
+				"Allow values: %v", strings.Trim(strings.Join(DomainEventTypesSupported[webhookType], ", "), "[]")),
+		},
+		"handler": {
+			Type: schema.TypeSet,
+			Required: true,
+			MinItems: 1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"url": {
+						Type:             schema.TypeString,
+						Required:         true,
+						ValidateDiagFunc: validation.ToDiagFunc(validation.All(validation.IsURLWithHTTPorHTTPS, validation.StringIsNotEmpty)),
+						Description:      "Specifies the URL that the Webhook invokes. This will be the URL that Artifactory will send an HTTP POST request to.",
+					},
+					"secret": {
+						Type:             schema.TypeString,
+						Optional:         true,
+						ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotEmpty),
+						Description:      "Secret authentication token that will be sent to the configured URL.",
+					},
+					"proxy": {
+						Type:             schema.TypeString,
+						Optional:         true,
+						ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotEmpty),
```

Comment:
Should we put a validator of "not a url" so people don't try it? What happens if you give it an invalid key?